testing/acceptance_test.py

```diff
@@ -1389,3 +1392,54 @@ def test_boo(self):
     )
     result = pytester.runpytest_subprocess()
     result.stdout.fnmatch_lines("*1 passed*")
+
+
+def test_issue_9765(pytester: Pytester) -> None:
+    """Reproducer for issue #9765 on Windows
+
+    https://github.com/pytest-dev/pytest/issues/9765
+    """
+    Path("setup.py").write_text(
+            textwrap.dedent("""
+            from setuptools import setup
+
+            if __name__ == '__main__':
+                setup(name='my_package')
+        """
+                            )
+    )
+    Path("pytest.ini").write_text(
+        textwrap.dedent("""
+            [pytest]
+            addopts = "-p my_package.plugin.my_plugin"
+        """
+                        )
+    )
+    pkg = pytester.mkpydir("my_package")
+    pkg.joinpath("conftest.py").write_text("")
+    pkg.joinpath("test_foo.py").write_text("def test(): pass")
+
+    plugin_dir = pytester.mkpydir("my_package/plugin")
+    plugin_dir.joinpath("my_plugin.py").write_text(
+        textwrap.dedent("""
+    import pytest
+
+    def pytest_configure(config):
+
+        class SimplePlugin:
+            @pytest.fixture(params=[1, 2, 3])
+            def my_fixture(self, request):
+                yield request.param
+
+        config.pluginmanager.register(SimplePlugin())
+    """
+        )
+    )
+
+    pytester.run(sys.executable, "setup.py", "develop")
```

Comment:
setup.py develop has been deprecated due to numerous issues with it 

does the issue only happen with legacy editables, or does it also happen with modern editables?

Reply:
I know but this is the we originally had the issue in scikit-learn, and we manage to put together a reproducer. This only happens with legacy editable install `python setup.py develop` or its pip equivalent `pip install --no-use-pep517 -e .`

Note that other projects (scipy and jdaviz for now but I could magine others) seem to have a very closely related problems due to normcase path difference on Windows. This does not seem to involve legacy editable installs.

Note although this is somewhat deprecated, we still recommend to use `pip install --no-use-pep517` for scikit-learn contributors. The incremental builds do not seem to work with the non-legacy installs so you rebuild everything from scratch. If you really want to know more, look at for example https://github.com/scikit-learn/scikit-learn/pull/27960